nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/test/java/org/apache/nifi/processors/kudu/TestPutKudu.java

```diff
@@ -366,54 +368,82 @@ public void testUpdateFlowFiles() throws Exception {
 
     @Test
     public void testBuildRow() {
-        buildPartialRow((long) 1, "foo", (short) 10);
+        buildPartialRow((long) 1, "foo", (short) 10, "id", "id", false);
     }
 
     @Test
     public void testBuildPartialRowNullable() {
-        buildPartialRow((long) 1, null, (short) 10);
+        buildPartialRow((long) 1, null, (short) 10, "id", "id",  false);
     }
 
     @Test(expected = IllegalArgumentException.class)
     public void testBuildPartialRowNullPrimaryKey() {
-        buildPartialRow(null, "foo", (short) 10);
+        buildPartialRow(null, "foo", (short) 10, "id", "id",  false);
     }
 
     @Test(expected = IllegalArgumentException.class)
     public void testBuildPartialRowNotNullable() {
-        buildPartialRow((long) 1, "foo", null);
+        buildPartialRow((long) 1, "foo", null, "id", "id", false);
     }
 
-    private void buildPartialRow(Long id, String name, Short age) {
+    @Test
+    public void testBuildPartialRowLowercaseFields() {
+        PartialRow row = buildPartialRow((long) 1, "foo", (short) 10, "id", "ID", true);
+        row.getLong("id");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testBuildPartialRowLowercaseFieldsFalse() {
+        PartialRow row = buildPartialRow((long) 1, "foo", (short) 10, "id", "ID", false);
+        row.getLong("id");
+    }
+
+    @Test
+    public void testBuildPartialRowLowercaseFieldsKuduUpper() {
+        PartialRow row = buildPartialRow((long) 1, "foo", (short) 10, "ID", "ID", false);
+        row.getLong("ID");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testBuildPartialRowLowercaseFieldsKuduUpperFail() {
+        PartialRow row = buildPartialRow((long) 1, "foo", (short) 10, "ID", "ID", true);
+        row.getLong("ID");
+    }
+
+    private PartialRow buildPartialRow(Long id, String name, Short age, String kuduIdName, String recordIdName, Boolean lowercaseFields) {
         final Schema kuduSchema = new Schema(Arrays.asList(
```

Comment:
@kjmccarthy nit: can you keep the original indent

Reply:
Fixed